### PR TITLE
core: return result for xhtml, but with warning

### PR DESCRIPTION
**Pull request by jckr**
**Summary**

This PR is in response to #11482. Currently, we can't run audits on webpages which have the mime type "application/xhtml+xml" - we get nothing. With this PR, we'll get a warning but the audit will run. 

**Related Issues/PRs**

### CODE REVIEW

File: lighthouse-core/gather/gather-runner.js [hunk 1]
```diff
@@ -39,6 +39,11 @@ const UIStrings = {
   warningSlowHostCpu: 'The tested device appears to have a slower CPU than  ' +
   'Lighthouse expects. This can negatively affect your performance score. Learn more about ' +
   '[calibrating an appropriate CPU slowdown multiplier](https://github.com/GoogleChrome/lighthouse/blob/master/docs/throttling.md#cpu-throttling).',
+
+  /**
+   * @description Warning that the page is using the xhtml mime type
+   */
+  warningXhtml: 'The page is using an XHTML mime type.',
```
Comment (connorjclark) [inline lighthouse-core/gather/gather-runner.js:46]:
```suggestion
  warningXhtml: 'The page MIME type is XHTML: Lighthouse does not explicitly support this document type.',
```

File: lighthouse-core/gather/gather-runner.js [hunk 2]
```diff
@@ -48,6 +53,10 @@ const UIStrings = {
  */
 const SLOW_CPU_BENCHMARK_INDEX_THRESHOLD = 1000;
 
+// MIME types are case-insenstive but Chrome normalizes MIME types to be lowercase.
```
Comment (connorjclark) [inline lighthouse-core/gather/gather-runner.js:56]:
```suggestion
// MIME types are case-insensitive but Chrome normalizes MIME types to be lowercase.
```

File: lighthouse-core/gather/gather-runner.js [hunk 3]
```diff
@@ -298,9 +306,14 @@ class GatherRunner {
     // Example: `DNS_FAILURE` is better than `NO_FCP`.
     if (networkError) return networkError;
 
-    // Error if page is not HTML.
-    if (nonHtmlError) return nonHtmlError;
-
+    if (finalRecord?.mimeType === XHTML_MIME_TYPE) {
```
Comment (connorjclark) [inline lighthouse-core/gather/gather-runner.js:309]: @brendankenny I _think we can use `?.` but want to double check. Just shouldn't in the report, right?

**Review by connorjclark — CHANGES_REQUESTED** [review]

**Comment by connorjclark** [comment]
fyi we've have many CI failures lately, you'll need to do a `git pull && git merge origin/master`.

File: lighthouse-core/gather/gather-runner.js (hunk 3 above)
Comment (brendankenny) [reply lighthouse-core/gather/gather-runner.js:309]: > I _think_ we can use `?.` but want to double check. Just shouldn't in the report, right?

not yet, we're still waiting for the switch to node 14 minimum

**Comment by simonbesters** [comment]
Any chance this PR is going to be picked up anytime soon? Would really help me. Thank you.

File: lighthouse-core/gather/gather-runner.js (hunk 3 above)
Comment (connorjclark) [reply lighthouse-core/gather/gather-runner.js:309]: (we can now use this syntax)

**Review by connorjclark — APPROVED** [review]

**Comment by ForbiddenEra** [comment]
> @jckr can you give us a signal if you'd like to continue work / address feedback? Otherwise, would someone else like to open a new PR doing the same?
> 
> > What issues are there?
> 
> As you can see from reading this PR...this PR is an external contribution and there is some unaddressed feedback. In the future, please refrain from commenting on every related issue with the same message-it's just spam and is not helpful.

Yeah; I hadn't intended on doing that but kept finding different related/similar ones and wasn't sure where to comment.

Regardless: I really appreciate you championing this and pushing it through so quickly, thanks!!

**Pull request merged**